### PR TITLE
Improvement/newtab

### LINKED ISSUE #181: Open new tab to the right of the active tab
Currently new files are opened in a tab at the far right of the tab bar.
Sometimes you just want to have a quick look at another file, may to just quickly copy some lines, and immediately close the file again. Unfortunately, you're now at the end of the tab bar and - if you have many files open - need to scroll back to the left in the tab bar and find the file you were at before.

If the new file would not open at the right end of the tab bar, but to the right of the currently active tab, you would automatically end up at the original tab when you close the newly opened tab.

### PR DESCRIPTION
**Pull request by jackburton79**
Open new tabs to the right of the currently selected one.
Fixes #181

### CODE REVIEW

File: src/ui/GenioWindow.cpp [hunk 1]
```diff
@@ -1705,40 +1705,40 @@ GenioWindow::_FileOpen(BMessage* msg)
...
 		noticeMessage.AddString("file_name", editor->FilePath());
 		SendNotices(MSG_NOTIFY_EDITOR_FILE_OPENED, &noticeMessage);
 
-		LogInfo("File open: %s [%d]", editor->Name().String(), fTabManager->CountTabs() - 1);
+		LogInfo("File open: %s [%d]", editor->Name().String(), newIndex - 1);
 	}
 
-	// If at least 1 item or more were added select the first
-	// of them.
-	if (nextIndex < fTabManager->CountTabs())
```
Comment (Freaxed) [inline src/ui/GenioWindow.cpp:1740]: Aren't we going to support this anymore? Imagine you drop 5 files in the editor, in the current version you select the first file on the left, in this change the last one on the right. Same when you open the application and you automatically reaload the open files.. 
Not sure if I care much to keep it to be honest.. for the dropped one I don't care for the ones loaded at startup maybe the right solution should be to also reload what was the selected one and reselect it.. but it's a different feature..

**Comment by jackburton79** [comment]
How do I reproduce this?

Il ven 2 feb 2024, 12:30 Freaxed ***@***.***> ha scritto:

**Comment by Freaxed** [comment]
first test: open genio, close all the open files, drag from outside two file. In the current version the fist one is selected in this branch version the second one.
Secondo test: open 3 files in genio. select the middle one. close genio. reopen genio: the middle file should be selected (current version) with this branch version the last one is selected

**Comment by jackburton79** [comment]
Thanks. Will fix this.

**Comment by jackburton79** [comment]
> first test: open genio, close all the open files, drag from outside two file. In the current version the fist one is selected in this branch version the second one. Secondo test: open 3 files in genio. select the middle one. close genio. reopen genio: the middle file should be selected (current version) with this branch version the last one is selected

Should be fixed

File: src/ui/GenioWindow.cpp [hunk 2]
```diff
@@ -1705,41 +1711,43 @@ GenioWindow::_FileOpen(BMessage* msg)
...
 		selectTabInfo.AddBool("caret_position", true);
-		int32 index = fTabManager->CountTabs();
+		int32 index = fTabManager->SelectedTabIndex() + 1;
 		Editor* editor = _AddEditorTab(&ref, index, &selectTabInfo);
 
+		LogTrace("New index: %d, selected index: %d", index, fTabManager->SelectedTabIndex());
+
 		// TODO: using assert() is not nice, try to handle
 		// this gracefully if possible
 		assert(index >= 0);
```
Comment (nexus6-haiku) [inline src/ui/GenioWindow.cpp:1721]: While you are there, Should these be turned into regular if statements?

Comment (jackburton79) [reply src/ui/GenioWindow.cpp]: done

**Pull request merged**